test/AWS_Tests/src/S3_Spec.enso

```diff
@@ -562,6 +567,17 @@ add_specs suite_builder =
             Panic.with_finalizer s3_link.delete <|
                 s3_link.read . should_equal "Hello WORLD!"
 
+        s3_to_s3_datalinks = Symlink_Test_Setup.make (my_writable_dir / "target-link-dir-1") (my_writable_dir / "datalink-dir-1") (create_data_link_to_s3 test_credentials) create_target_directory=x->x create_datalink_location_directory=x->x
```

Comment:
If I understand this correctly, you override the default `create_directory` here with a no-op `x->x` for S3. I just read the explanation in #9704. Would it make sense for now to implement `S3_File.create_directory` as a no-op? So that this code (and user code) can call this method without having to handle it separately for S3.

I realize that this might not behave as expected with the `list` method, but that is expected for S3. The empty object trick can be done, but I don't think it's used that often.

Reply:
I like that, although I'm not fond of `exists` being `False` after `create_directory`.

What about a compromise - we make `create_directory` a no-op but it reports a warning explaining this S3 peculiarity?

Reply:
Yes, perfect.